lib/guardian/bookmark_guardian.rb

```diff
@@ -9,7 +9,11 @@ def can_edit_bookmark?(bookmark)
     @user == bookmark.user
   end
 
-  def can_create_bookmark?(bookmark)
-    can_see_topic?(bookmark.topic)
+  def can_see_bookmarkable?(bookmark)
+    if !SiteSetting.use_polymorphic_bookmarks?
+      return self.can_see_post?(bookmark.post)
+    end
+
+    bookmark.registered_bookmarkable.can_see?(self, bookmark)
```

Comment:
```suggestion
  def can_see_bookmarkable?(bookmark)
    if SiteSetting.use_polymorphic_bookmarks?
      bookmark.registered_bookmarkable.can_see?(self, bookmark)
    else
      self.can_see_post?(bookmark.post)
    end
```

Reply:
Will slightly change, but I still prefer the early return pattern, so will do this:

```ruby
if SiteSetting.use_polymorphic_bookmarks?
  return bookmark.registered_bookmarkable.can_see?(self, bookmark)
end

self.can_see_post?(bookmark.post)
```